lib/Sema/CSDiagnostics.h

```diff
@@ -667,6 +667,22 @@ class AssignmentFailure final : public FailureDiagnostic {
   Optional<OverloadChoice> getMemberRef(ConstraintLocator *locator) const;
 };
 
+class NonClassTypeToAnyObjectConversionFailure : public FailureDiagnostic {
+  Type RawFromType, RawToType;
+  ConstraintLocator *RawLocator;
```

Comment:
`FailureDiagnostic` already saves the locator it got passed in so you don't need to safe it this way.